src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/DryRunNotificationService.kt

```diff
@@ -53,39 +48,26 @@ class DryRunNotificationService(
         log.info("dryRunSendNotifications: Sent notification (${group.value.size} lines) for ${group.key}")
       }
     log.info("dryRunSendNotifications: Finished sending notifications")
-    val duration = System.currentTimeMillis() - start
-    telemetryClient.trackEvent(
-      "dryRunSendNotifications",
-      ImmutableMap.of(
-        "unprocessedNotifications", "${unprocessedNotifications.size}",
-        "durationMillis", duration.toString(),
-        "duration", DurationFormatUtils.formatDuration(duration, "HH:mm:ss")
-      ),
-      null
-    )
   }
 
   @Transactional
   fun dryRunNotifications(region: Int) {
-    val start = System.currentTimeMillis()
     log.info("dryRunNotifications region: $region")
 
     val details = csrClient.getModified(region)
 
-    log.debug("dryRunNotifications region: $region found ${details.size} cmd_notifications details")
-
     val cutoffTime = LocalDateTime.now(clock).minusMinutes(CUTOFF_MINUTES)
 
     val usersWithoutRecentActivity = details
-      .filter { it.quantumId != null && it.shiftModified != null }
       .groupingBy { it.quantumId }
       .aggregate(::latestShiftModified)
-      .filter { it.value.shiftModified!!.isBefore(cutoffTime) }
+      .filter { it.value.shiftModified?.isBefore(cutoffTime) ?: true }
```

Comment:
if all shiftModified's for a user are null, count it as without recent activity. The records will then get deleted from the polling table